libsolidity/codegen/LValue.cpp

```diff
@@ -216,22 +216,38 @@ void StorageItem::retrieveValue(SourceLocation const&, bool _remove) const
 void StorageItem::storeValue(Type const& _sourceType, SourceLocation const& _location, bool _move) const
 {
 	CompilerUtils utils(m_context);
+	solAssert(m_dataType, "");
```

Comment:
@pirapira the stack element might not have been converted (because when I assert `_sourceType == *m_dataType` it fails).